forwardproxy.go

```diff
@@ -232,7 +236,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyht
 		return caddyhttp.Error(http.StatusProxyAuthRequired, authErr)
 	}
 
-	if r.ProtoMajor != 1 && r.ProtoMajor != 2 {
+	if r.ProtoMajor != 1 && r.ProtoMajor != 2 && r.ProtoMajor != 3 {
```

Comment:
Did you test this with HTTP/3? I haven't, and I'd be surprised if this works with H3.